shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -675,103 +684,208 @@ mod cassandra_protocol_tests {
     573730010000e736368656d615f76657273696f6e000c0006746f6b656e730022000d00000000"
     );
 
-    fn test_frame(codec: &mut CassandraCodec2, raw_frame: &[u8]) {
-        let message = codec
+    fn test_frame_codec_roundtrip(
+        codec: &mut CassandraCodec2,
+        raw_frame: &[u8],
+        expected_messages: Vec<Message>,
+    ) {
+        // test decode
+        let decoded_messages = codec
             .decode(&mut BytesMut::from(raw_frame))
             .unwrap()
             .unwrap();
+        assert_eq!(decoded_messages, expected_messages);
 
+        // test encode round trip
         let mut dest = BytesMut::new();
-        codec.encode(message, &mut dest).unwrap();
+        codec.encode(decoded_messages, &mut dest).unwrap();
         assert_eq!(raw_frame, &dest);
     }
 
-    #[test]
-    fn test_startup_codec() {
+    fn new_codec() -> CassandraCodec2 {
         let mut pk_map = HashMap::new();
         pk_map.insert("test.simple".to_string(), vec!["pk".to_string()]);
         pk_map.insert(
             "test.clustering".to_string(),
             vec!["pk".to_string(), "clustering".to_string()],
         );
-        let mut codec = CassandraCodec2::new(pk_map, false);
-        test_frame(&mut codec, &STARTUP_BYTES);
+        CassandraCodec2::new(pk_map, false)
+    }
+
+    #[test]
+    fn test_startup_codec() {
+        let mut codec = new_codec();
+        let messages = vec![Message {
+            details: MessageDetails::Unknown,
+            modified: false,
+            original: RawFrame::Cassandra(Frame {
+                version: Version::Request,
+                flags: vec![],
+                opcode: Opcode::Startup,
+                stream: 0,
+                body: vec![
```

Comment:
we build binary blobs now two ways in our tests. One using the vec macro and one using the hex macro. We should pick one and keep it consistent.

If we want to go with vec! for all things then we can get rid of one more dependency

Reply:
That's a good point, I had just copied the vec from the output of the assert_eq failure.
But using hex! for this seems like the right approach.